data/basics/units.go

```diff
@@ -108,6 +108,12 @@ func (a MicroAlgos) MsgIsZero() bool {
 	return a.Raw == 0
 }
 
+// MicroAlgosMaxSize returns maximum possible msgp encoded size of MicroAlgos in bytes.
+// It is expected by msgp generated MaxSize functions
+func MicroAlgosMaxSize() (s int) {
```

Comment:
Why is this needed? MicroAlgos has one field of type uint64.

Reply:
Same reason as the previous conversations we've had: `msgp` when it sees a type defined outside of the current file it's parsing it assumes that it has this function defined. MicroAlgos is an example of a struct that we generate fully manually and don't use msgp for but we still need this method and the ones above for the rest of the msgp generated functions to work since they call them when they are Marshalling/Unmarshalling themselves or checking their own MaxSize